signing/secp256k1/api/src/main/java/tech/pegasys/signers/secp256k1/api/SignerProvider.java

```diff
@@ -15,12 +15,13 @@
 import java.security.interfaces.ECPublicKey;
 import java.util.Optional;
 import java.util.Set;
+import java.util.function.Function;
 
 public interface SignerProvider {
 
-  Optional<Signer> getSigner(ECPublicKey publicKey);
+  Optional<Signer> getSigner(SignerIdentifier publicKey);
 
-  Set<ECPublicKey> availablePublicKeys();
+  Set<ECPublicKey> availablePublicKeys(Function<ECPublicKey, SignerIdentifier> identifierFunction);
```

Comment:
This makes the API a bit odd and this function doesn't seem related to returning available public keys. I can see why it's necessary though from how it's used in the MultiKeySignerProvider.

Reply:
Yes, I couldn't think of another way to validate that the loaded signers matches the "identifier" in filename without this approach.